botnim/vector_store/vector_store_base.py

```diff
@@ -21,13 +22,21 @@ def env_name_slug(self, name):
             name += '__dev'
         return name
 
-    def vector_store_update(self, context, replace_context):
+    def vector_store_update(self, context, replace_context, extract_metadata=False):
         for context_ in context:
             context_name = context_['slug']
             vector_store = self.get_or_create_vector_store(context_, context_name, replace_context)
-            file_streams = collect_context_sources(context_, self.config_dir)
-            file_streams = [((fname if self.production else '_' + fname), f, t) for fname, f, t in file_streams]
-            file_names = [fname for fname, _, _ in file_streams]
+            file_streams = collect_context_sources(context_, self.config_dir, extract_metadata=extract_metadata)
```

Comment:
Again, a complete rewrite of a section of the code when all that was needed was to add a single argument to a single function. This time, also introducing a serious bug.